### ISSUE TITLE
Possible performance problem with combineLatest

### ISSUE THREAD

**Issue by zywind** [cyclejs/cycle-dom#53]
I am writing a program that renders the US county map and I'm using the nested dialogue structure.  But I found that when I try to combine the map with another constantly updating div, the performance took a huge hit. Here is the code:

```javascript
let mapDom$ = paths$.map(({counties, states, borders}) => {
    console.log('render');
    const c = svg('g', counties.map( ({id, d}) => svg('path', {key: id, d: d, fill: 'white'})));
    const s = createStates(states);
    const b = svg('path', {d: borders, class: mapStyles.statesborder});

    return svg('svg', {width: window.innerWidth, height: window.innerHeight}, [c, s, b]);
  });

  let intervalDOM$ = Rx.Observable.interval(1000)
    .map( i =>
      div(i.toString())
      );

  let vtree$ = Rx.Observable.combineLatest(mapDom$, intervalDOM$, (mapDom, intervalDom) => div([mapDom, intervalDom]));
```

The problem seems to be combineLatest. I think it might be making a copy of the mapDOM every time the intervalDOM updates.  Is there a solution to this?

**Comment by zywind** [comment]
So it turns out the problem was not in combineLatest but in virtual-dom. The evidence is that the following code does not slow down
```javascript
combineLatest(mapDom$, intervalDOM$, (mapDom, intervalDom) => intervalDom)
```
But if the function returns a mapDom, which is a huge vtree, then the application slows down. So the problem is likely caused by diffing the huge vtree.

**Comment by staltz** [comment]
Check out virtual-dom thunks: https://github.com/Matt-Esch/virtual-dom/blob/master/docs/thunk.md

**Comment by zywind** [comment]
Does thunk work in cycle-dom? I just tried that simple ConstantlyThunk example with cyclejs, and it doesn't seem to show anything. The root tree is blank.

**Comment by staltz** [comment]
There is no reason why it shouldn't work. Can you provide a minimal example code to reproduce the bug?

**Comment by zywind** [comment]
Here is a minimal example. I don't see anything under the root node.  Maybe I'm using thunk in a wrong way?

```javascript
import {Rx, run} from '@cycle/core';
import {h, makeDOMDriver} from '@cycle/dom';

let ConstantlyThunk = function(greeting){
  this.greeting = greeting;
};
ConstantlyThunk.prototype.type = "Thunk";
ConstantlyThunk.prototype.render = function(previous) {
  if (previous && previous.vnode) {
    return previous.vnode;
  } else {
    return h('div', ["Constantly "+ this.greeting]);
  }
}

function main({DOM}) {
  let vtree$ = Rx.Observable.return(true)
    .map( () => new ConstantlyThunk('hello'));

  return {
    DOM: vtree$
  };
}

run(main, {
  DOM: makeDOMDriver('#root')
});
```

**Comment by staltz** [comment]
You're right, there was a bug in Cycle DOM, I'll fix it asap.

**Comment by staltz** [comment]
Released v7.1.0, reopen this issue if you still have problems when using thunks.
